coderedcms/templatetags/coderedcms_tags.py

```diff
@@ -40,6 +40,12 @@ def coderedcms_version():
 def generate_random_id():
     return ''.join(random.choice(string.ascii_letters + string.digits) for n in range(20))
 
+@register.simple_tag
+def twitter_card_image(self):
```

Comment:
`self` is a special keyword in Python. In the context of a python file, `self` refers to the actual file. Inside of a `class`, self would refer to the class (it's like `this` in JavaScript or other languages). So in this context, `self` refers to `coderedcms_tags`. Whereas in the template, `self` was referring to the page model.

Try using a different variable name, such as `page` (since you are expecting a page object).